### ISSUE TITLE
[display] module-private-type shows with _ on @type

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#5627]
Another implementation detail:

![](http://i.imgur.com/wtzsCm1.png)

```haxe
class Main {
	public static function main() {}
}


private enum TestEnum {
	Value;
}
```

**Comment by Simn** [comment]
This one is a bit annoying because we use `s_type` for type printing. I don't really want to write a new printing function just for this case, so I suggest we wait until @nadako finished his JSON output thing.

**Comment by nadako** [comment]
I guess I should rebase that branch and make it work for 3.3...

**Comment by Gama11** [comment]
Do it! :)

**Comment by Gama11** [comment]
It seems that right now, even with JSON output we don't get enough info to print it properly?

```json
"type": {
    "kind": "TEnum",
    "args": {
        "path": {
            "pack": [
                "_Main"
            ],
            "name": "TestEnum"
        },
        "params": []
    }
}
```

**Comment by Simn** [comment]
This would be solved as a subset of your "don't show dot paths if not necessary" idea.

**Comment by Gama11** [comment]
I'm not sure it would be - there are cases where we still want to show the full dot path even if imported. E.g. we do that in completion right now:

![](https://i.imgur.com/tRhyiTk.png)

We do that because otherwise you have no way to tell what package the type comes from if imported (without checking the imports).

**Comment by Simn** [comment]
Is this still a problem with the "origin" information?

**Comment by Gama11** [comment]
Not sure how that's related - origin is for fields only, this is also an issue for types (see the screenshot in my previous comment).

**Comment by Simn** [comment]
It's still not clear to me what information you need here. I'm seeing this:

```
                "type": {
                    "kind": "TEnum",
                    "args": {
                        "path": {
                            "pack": [
                                "_Main"
                            ],
                            "moduleName": "Main",
                            "typeName": "TestEnum",
                            "importStatus": 0
                        },
                        "params": []
                    }
                }
```

What should be added there?

**Comment by Gama11** [comment]
Shouldn't `pack` simply be empty here?

**Comment by Simn** [comment]
That's a bit of a general question that also affects some features like `@:access`. Our internal data represents the generator-facing information, but here it would be better to show the user-facing one. I have to see if this can be dealt with easily.

**Comment by nadako** [comment]
@Gama11 is this a non-issue with latest JSON-RPC protocol?

**Comment by Gama11** [comment]
No, this is still an issue - we still leak these implementation details.

**Comment by nadako** [comment]
Well, leaking generated modules is a more general issue (e.g. see also #9214). The question is if vshaxe works this around in some hacky way it we have enough proper data to show correct hints?

**Comment by Gama11** [comment]
I don't think we have enough info to work around it, at least not in a way that works 100% of the time...
